src/main/java/com/influxdb/v3/client/internal/FlightSqlClient.java

```diff
@@ -133,14 +138,38 @@ Stream<VectorSchemaRoot> execute(@Nonnull final String query,
         Ticket ticket = new Ticket(json.getBytes(StandardCharsets.UTF_8));
         FlightStream stream = client.getStream(ticket, callOptionArray);
         FlightSqlIterator iterator = new FlightSqlIterator(stream);
+        addToAutoCloseable(stream);
 
         Spliterator<VectorSchemaRoot> spliterator = Spliterators.spliteratorUnknownSize(iterator, Spliterator.NONNULL);
         return StreamSupport.stream(spliterator, false).onClose(iterator::close);
     }
 
+    private synchronized void addToAutoCloseable(@Nonnull final AutoCloseable closeable) {
+        // need to occasionally clean up references to closed streams
+        // in order to ensure memory can get freed.
+        if (autoCloseables.size() > AUTOCLOSEABLE_CHECK_LIMIT) {
+            LOG.debug("checking to cleanup stale flight streams from {} known streams", autoCloseables.size());
+
+            ListIterator<AutoCloseable> iter = autoCloseables.listIterator();
+            while (iter.hasNext()) {
+                AutoCloseable autoCloseable = iter.next();
+                if (CLOSEABLE_CLOSED_LEDGER.get(autoCloseable)) {
+                    LOG.debug("removing closed stream {}", autoCloseable);
+                    CLOSEABLE_CLOSED_LEDGER.keySet().remove(autoCloseable);
+                    iter.remove();
+                }
```

Comment:
A more effecient way to remove:
```suggestion
                if (CLOSEABLE_CLOSED_LEDGER.remove(autoCloseable, true)) {
                    LOG.debug("removing closed stream {}", autoCloseable);
                    iter.remove();
                }               
```